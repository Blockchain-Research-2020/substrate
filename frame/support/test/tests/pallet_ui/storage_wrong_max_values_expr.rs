#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::{Hooks, Twox64Concat, StorageMap};
	use frame_system::pallet_prelude::BlockNumberFor;

	#[pallet::config]
	pub trait Config: frame_system::Config {}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::hooks]
	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}

	#[pallet::call]
	impl<T: Config> Pallet<T> {}

	#[pallet::storage(max_values = 4u64)]
	type Foo<T> = StorageMap<_, Twox64Concat, u32, u32>;
}

fn main() {
}
