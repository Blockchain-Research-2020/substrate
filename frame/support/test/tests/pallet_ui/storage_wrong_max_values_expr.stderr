error[E0308]: mismatched types
  --> $DIR/storage_wrong_max_values_expr.rs:18:33
   |
9  |     #[pallet::pallet]
   |               ------ expected due to this
...
18 |     #[pallet::storage(max_values = 4u64)]
   |                                    ^^^^ expected `u32`, found `u64`
   |
help: change the type of the numeric literal from `u64` to `u32`
   |
18 |     #[pallet::storage(max_values = 4u32)]
   |                                    ^^^^
